src/client/datascience/jupyter/pythonVariableRequester.ts

```diff
@@ -14,6 +13,28 @@ import { JupyterDataRateLimitError } from '../errors/jupyterDataRateLimitError';
 import { executeSilently } from './kernels/kernel';
 import { IKernel } from './kernels/types';
 
+type DataFrameSplitFormat = {
+    index: (number | string)[];
+    columns: string[];
+    data: Record<string, unknown>[];
+};
+
+export function parseDataFrame(df: DataFrameSplitFormat) {
+    const rowIndexValues = df.index;
+    const columns = df.columns;
+    const rowData = df.data;
+    const data = rowData.map((row, index) => {
+        const rowData: Record<string, unknown> = {
+            index: rowIndexValues[index]
+        };
+        columns.forEach((column, columnIndex) => {
+            rowData[column] = row[columnIndex];
+        });
+        return rowData;
+    });
+    return { data };
+}
```

Comment:
This is the magic, this is where we convert the `split` (series) data format into a table (the same structure we used to get when we called `to_json with table`